src/core/prompts/tools/native-tools/index.ts

```diff
@@ -25,31 +26,44 @@ import editFile from "./kilocode/edit_file"
 export { getMcpServerTools } from "./mcp_server"
 export { convertOpenAIToolToAnthropic, convertOpenAIToolsToAnthropic } from "./converters"
 
-export const nativeTools = [
-	// kilocode_change start
-	searchAndReplace,
-	deleteFile,
-	editFile,
-	// todo:
-	// condenseTool,
-	// newRuleTool,
-	// reportBugTool,
-	// kilocode_change end
-	askFollowupQuestion,
-	attemptCompletion,
-	browserAction,
-	codebaseSearch,
-	executeCommand,
-	fetchInstructions,
-	generateImage,
-	insertContent,
-	listCodeDefinitionNames,
-	listFiles,
-	newTask,
-	read_file,
-	runSlashCommand,
-	searchFiles,
-	switchMode,
-	updateTodoList,
-	writeToFile,
-] satisfies OpenAI.Chat.ChatCompletionTool[]
+/**
+ * Get native tools array, optionally customizing based on settings.
+ *
+ * @param partialReadsEnabled - Whether to include line_ranges support in read_file tool (default: true)
+ * @returns Array of native tool definitions
+ */
+export function getNativeTools(partialReadsEnabled: boolean = true): OpenAI.Chat.ChatCompletionTool[] {
+	return [
+		// kilocode_change start
+		searchAndReplace,
+		deleteFile,
+		editFile,
+		// todo:
+		// condenseTool,
+		// newRuleTool,
+		// reportBugTool,
+		// kilocode_change end
+		accessMcpResource,
+		// apply_diff_single_file, TODO: fix this
```

Comment:
```suggestion
		deleteFile,
		editFile,
		// todo:
		// condenseTool,
		// newRuleTool,
		// reportBugTool,
		// kilocode_change end
		accessMcpResource,
		apply_diff_single_file,
```

I think the implementation of searchAndReplace was removed?